mne/io/nirx/nirx.py

```diff
@@ -0,0 +1,190 @@
+# Authors: Robert Luke <mail@robertluke.net>
+#
+# License: BSD (3-clause)
+
+import configparser as cp
+import re as re
+import glob as glob
+import pandas as pd
+import numpy as np
+
+from ..base import BaseRaw
+from ..meas_info import create_info
+from ...utils import logger, verbose, fill_doc
+
+
+@fill_doc
+def read_raw_nirx(fname, preload=False, verbose=None):
+    """Reader for a NIRX fNIRS recording.
+    Parameters
+    ----------
+    fname : str
+        Path to the NIRX data folder.
+    %(preload)s
+    %(verbose)s
+    Returns
+    -------
+    raw : instance of RawNIRX
+        A Raw object containing NIRX data.
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+    return RawNIRX(fname, preload, verbose)
+
+
+@fill_doc
+class RawNIRX(BaseRaw):
+    """Raw object from a NIRX fNIRS file.
+    Parameters
+    ----------
+    fname : str
+        Path to the NIRX data folder.
+    %(preload)s
+    %(verbose)s
+    See Also
+    --------
+    mne.io.Raw : Documentation of attribute and methods.
+    """
+
+    @verbose
+    def __init__(self, fname, preload=False, verbose=None):
+        logger.info('Loading %s' % fname)
+
+        # Check if required files exist
+
+        file_dat = glob.glob(fname + '/*.dat')
+        file_evt = glob.glob(fname + '/*.evt')
+        file_hdr = glob.glob(fname + '/*.hdr')
+        file_inf = glob.glob(fname + '/*.inf')
+        file_set = glob.glob(fname + '/*.set')
+        file_tpl = glob.glob(fname + '/*.tpl')
+        file_wl1 = glob.glob(fname + '/*.wl1')
+        file_wl2 = glob.glob(fname + '/*.wl2')
+        file_cfg = glob.glob(fname + '/*config.txt')
+        file_mat = glob.glob(fname + '/*probeInfo.mat')
+
+        assert (len(file_dat) == 1), "Should be one dat file"
```

Comment:
`assert` statements are skipped completely when python is run with `python -O`, so these need to be proper checks. To make your life easier you could write a help function for this. Also in general we try to give helpful hints on error, so you could do for example:
```
def assert_one(x, msg):
     if len(x) != 1:
        raise RuntimeError(msg + ', got %d' %(len(x),))
```
or so.